tests/unit/test_task_runner.cpp

```diff
@@ -252,6 +252,83 @@ TEST(TaskRunner, TestDelayedTasksOrder)
     STRCMP_EQUAL("bac", str.c_str());
 }
 
+TEST(TaskRunner, TestCancelDelayedTasks)
+{
+    std::string              str;
+    otbr::TaskRunner         taskRunner;
+    otbr::TaskRunner::TaskId tid1, tid2, tid3;
+
+    tid1 = taskRunner.Post(std::chrono::milliseconds(10), [&]() { str.push_back('a'); });
+    tid2 = taskRunner.Post(std::chrono::milliseconds(20), [&]() { str.push_back('b'); });
+    tid3 = taskRunner.Post(std::chrono::milliseconds(30), [&]() { str.push_back('c'); });
+
+    CHECK(0 < tid1);
+    CHECK(tid1 < tid2);
+    CHECK(tid2 < tid3);
+
+    taskRunner.Cancel(tid2);
```

Comment:
Test also if we can successfully cancel the task after mainloop is started? We can do this by cancelling this task in a fourth task:

```
taskRunner.Post(std::chrono::milliseconds(10), [&]() { taskRunner.Cancel(tid2); });
```

or in a new Thread:
```
std::thread([&]() { sleep(10.0/1000); taskRunner.Cancel(tid2); });
```

Reply:
That's a good idea.

Reply:
Test enhanced.